components/RadioButtons.tsx

```diff
@@ -33,28 +32,34 @@ const RadioButtons = ({
 
   return (
     <>
-      {title && <h3>{title}</h3>}
       <fieldset
+        aria-labelledby={`button-rating-${title}`}
         className={styles.container}
         disabled={disabled}
-        onChange={
-          (e) =>
-            updateSelectedOption(parseInt((e.target as HTMLInputElement).value))
-          // eslint-disable-next-line react/jsx-curly-newline
-        }
+        onChange={(e) => {
+          const selectedValue = parseInt((e.target as HTMLInputElement).value)
+          updateSelectedOption(selectedValue)
+        }}
       >
+        {title && (
+          <legend id={`button-rating-${title}`}>
```

Comment:
I think you can remove these ids from the legend, per the note about `aria-labelledby`!